includes/reader-revenue/woocommerce/class-woocommerce-duplicate-orders.php

```diff
@@ -0,0 +1,247 @@
+					<?php echo esc_html__( 'There are some potentially duplicate transactions to review. Some of these might be intentional. Click this message to display the list of possible duplicates.', 'newspack-plugin' ); ?>
+				</summary>
+				<ul>
+					<?php foreach ( $existing_order_duplicates as $order_duplicates ) : ?>
+						<?php
+						if ( in_array( $order_duplicates['ids'], $dismissed_duplicates ) ) {
+							continue;}
+						?>
+						<li style="display: flex; align-items: center;">
+							<p style="margin: 0;">
+
+							<?php
+							ob_start();
+							?>
+								<a href="<?php echo esc_url( admin_url( 'edit.php?s=' . urlencode( $order_duplicates['email'] ) . '&post_type=shop_order' ) ); ?>"><?php echo esc_html( $order_duplicates['email'] ); ?></a>
+							<?php
+							$customer_email = ob_get_clean();
+
+							ob_start();
+							$order_ids = explode( ',', $order_duplicates['ids'] );
+							foreach ( $order_ids as $index => $order_id ) :
+								$order_url = admin_url( 'post.php?post=' . intval( $order_id ) . '&action=edit' );
+								?>
+									<a href="<?php echo esc_url( $order_url ); ?>"><?php echo esc_html( $order_id ); ?></a><?php echo ( $index < count( $order_ids ) - 1 ) ? ', ' : ''; ?>
+								<?php
+							endforeach;
+							$order_list = ob_get_clean();
+
+							printf(
+								/* translators: 1: customer email, 2: order amount, 3: orders date, 4: order IDs */
+								wp_kses_post( __( 'Customer %1$s made multiple orders of %2$s on %3$s. Orders: %4$s.', 'newspack-plugin' ) ),
+								wp_kses_post( $customer_email ),
+								wp_kses_post( \wc_price( $order_duplicates['amount'] ) ),
+								esc_html( date_i18n( get_option( 'date_format' ), strtotime( $order_duplicates['date'] ) ) ),
+								wp_kses_post( trim( $order_list ) )
+							);
+
+							$order_duplicates_id = implode( '-', $order_ids );
+						?>
+							</p>
+							<form method="post" style="display:inline; margin-left: 8px;">
+								<input type="hidden" name="dismiss_order_ids" value="<?php echo esc_attr( $order_duplicates['ids'] ); ?>">
+								<?php submit_button( __( 'Dismiss', 'newspack-plugin' ), 'small', 'dismiss_order', false, [ 'id' => 'dismiss_order_duplicates_' . $order_duplicates_id ] ); ?>
+							</form>
+						</li>
+					<?php endforeach; ?>
+				</ul>
+			</details>
+		</div>
+		<?php
+		if ( isset( $_POST['dismiss_order'] ) && isset( $_POST['dismiss_order_ids'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Missing
+			$dismissed_duplicates[] = $_POST['dismiss_order_ids']; // phpcs:ignore WordPress.Security.NonceVerification.Missing, WordPress.Security.ValidatedSanitizedInput.InputNotSanitized
+			update_option( self::DISMISSED_DUPLICATES_OPTION_NAME, $dismissed_duplicates );
+			// Refresh the page to reflect changes.
+			wp_safe_redirect( isset( $_SERVER['REQUEST_URI'] ) ? $_SERVER['REQUEST_URI'] : admin_url() );
+			exit;
+		}
+	}
+
+	/**
+	 * CLI handler to upsert the order duplicates with a specified timeframe.
```

Comment:
is it detect or upsert?

Reply:
Let's add a clearer description of what the command does
- search for duplicates and optionally store this info to be displayed in the admin panel

Reply:
Updated in 520f49c1367f00aebf272734c855943de2c7080a